### ISSUE TITLE
Locate missing descriptions

### ISSUE THREAD

**Issue by rclark** [rclark/grand-canyon-geology#2]
There are a number of units that lack descriptions -- they didn't match perfectly with the [NPS GRCA dataset](https://irma.nps.gov/App/Reference/Profile/2203644).

Many are close matches. Follow up with Jim to track these down.

**Comment by brandon-rhodes** [comment]
I was noticing this morning, as I prepare for a Tanner hike in a few weeks, that the wide expanses of turquoise Supai beneath the rim are labelled "IPMs Unknown". From a few papers online (like [this one](https://repository.asu.edu/attachments/178485/content/Darling_asu_0010E_16576.pdf)), it looks like the title "lower Supai group" would be accurate.

Even without a full description, it would be nice to have those strata labeled, and I'd be happy to offer a pull request — but I can't quite tell whether I'd be supposed to edit `all-units.csv`, or `descriptions.csv`, or another file; I can't tell from the finished web page which data comes from exactly which files.

Thanks again for maintaining this superb resource!

**Comment by rclark** [comment]
Hi @brandon-rhodes! 

The source of this dataset was a set of publications by [George Billingsley at the USGS](https://geomaps.wr.usgs.gov/arizona/) over the course of several years. Unit descriptions were also cross-referenced against [a NPS dataset](https://irma.nps.gov/DataStore/Reference/Profile/2203644) that helped me fill in some gaps in text descriptions and also unify some abbreviations and select regionally-appropriate color-schemes. All this said: I am no expert in this area. I didn't go much farther because I don't consider myself well-informed enough to really to a good job of deeper compilation.

Aside from that:

- Updating `all-units.csv` would be the where we'd want to improve on the unit compilations. Note that I'm not in a position that we could "split" any existing units in that document because it'd mean adding new contacts to the map itself.

- Even with that in place, there are some significant technical hurdles in the way of my making updates to this map. This is due to the trajectory of where Mapbox (a 3rd party service which serves the data behind this map) has taken their service -- many of the tools that I used to create this map are no longer supported. It just so happens that I work at Mapbox, so I may be able to find some "loopholes"... but due to some personal issues I am not going to be able to work on this in the near future (i.e. not until the fall/winter 2019).

Sorry about this sort of bummer response... but if you'd still like to submit a PR to suggest adjustments to the `all-units.csv` I would love to review and hunt for a way to make updates to the map when I can make the time. Being as detailed as you can about references for any change would be extremely welcome.

**Comment by brandon-rhodes** [comment]
As Ohio’s stay-at-home guidelines have left me a bit more time than usual for personal projects, I thought I would follow up about how things are going with the geology map — which I used several times over the past month or two, both before and after my first-ever hike to Clear Creek a few years ago. Obviously I didn’t yet get around to contributing descriptions for some of the formations that are missing them; I guess I worried that, if you can no longer update the site, the effort would never make it on to the map itself.

So I wanted to check in and find out whether you'd recovered the ability to update the map. If not, and I had a weekend to play with maps, would it make sense for me to explore how another map API might let a newer version of the data get up and in front of folks who would use it.

I hope you've been enjoying Mapbox in the meantime and that you're safe amidst the current events.

**Comment by rclark** [comment]
Hi again! 

I think that for the time being, if you're looking for a side-project, I'd suggest you see what you can do with another map API (or with Mapbox's current tooling) after pulling down the geojson and description data in this repository. Hurts me to say it, but I'm still not in a great position to be able to drive a new version of the site.

I'm considering pulling all the map tiles for the existing map into a personal S3 bucket before Mapbox completely deprecates the APIs that the site is currently using to expose them. That'd keep the site online, but it wouldn't allow for modifications until I have a chance to try building out the vector tileset again using the current mechanisms available to me.

For context: the big problem is that while the Mapbox API lets me upload GeoJSON and have them make tiles for me, that pipeline will drive simplification of the lines/polygons that isn't acceptable for the topology of a geologic map. There's a new service I've been meaning to explore that may give me the control that this map needs, but I just haven't had a chance to explore it deeply.

**Comment by brandon-rhodes** [comment]
Are the files `geolines.geojson` and `geopolys.geojson` derived from even more fundamental data sources, or should I treat them as the ground-truth (for this project at least) about which rocks are where?

**Comment by brandon-rhodes** [comment]
After a few hours of work learning how Mapnik works, I have a small script which produces:

![image](https://user-images.githubusercontent.com/166162/79622533-a0b05c00-80e5-11ea-8d62-7e76292c344c.png)

Does that appear to have the level of detail that was lacking when you had a service do the rendering instead? The script that produced the above map takes about ~33s on my desktop machine, much of it loading in the two big JSON files.

1. It looks like your own maps also included a USGS quad layer showing things like roads — or, is that somewhere in this data set and I just haven't found out how to render it yet?
2. Obviously I'll have to find out how to produce not just one map but a big collection of tiles.
3. And I'm not even sure how it works that layers are clickable. I should take a look at the JS to see how that works.

But in any case, it looks straightforward to render. Let me know if the level of detail looks good or whether it's suffering from the problems that hit you before.

**Comment by rclark** [comment]
Nice work! Figuring out mapnik is no small feat. To answer some of your questions:

1. The ancillary data shown in the existing site are Mapbox layers that aren't included in this repository. The site was built by layer several datasources, and only the geologic ones are included here. I also did some blending with a hillshade layer for effect. All of the mapnik configuration that I used to do this should be laying around somewhere, I'll look it up today.

2. It is the tiling step which can cause degradations in precision. It looks to me like the approach you took in your script does not do any simplification, which is nice. I suspect that you're also using it to output some kind of image format, png or so? The "easy" tiling approaches at my disposal (Mapbox does the processing for you) create tiled **vector data**, and the process of generating that derivative vector data is where simplification causes problems. 

3. The clickable stuff is driven in JS on the site itself [by this code here](https://github.com/rclark/grand-canyon-geology/blob/fa032139354f5aa2070c228390a8906eb6599d87/scripts/gcg.js#L47-L50).

**Comment by brandon-rhodes** [comment]
> All of the mapnik configuration that I used to do this should be laying around somewhere, I'll look it up today.

Thanks!

> It looks to me like the approach you took in your script does not do any simplification, which is nice. I suspect that you're also using it to output some kind of image format, png or so?

Correct, there's no simplification! I just take the raw polygons and lines and burn up the CPU to render them at full precision on whatever scale image I want. It's kind of fun, I so rarely get to challenge my CPU these days.

And, yes, I'm at the moment just writing a single image file as a test of loading data and getting a projection set up successfully. My (still rather messy) script:

https://github.com/brandon-rhodes/grand-canyon-geology/blob/try-mapnik/build/test.py

— does the save with:

```
    mapnik.render_to_file(m, 'world.png', 'png')
```

> The clickable stuff is driven in JS on the site itself …

Thanks, that will give me something to look at once I have tiles rendered.

**Comment by rclark** [comment]
@brandon-rhodes to get back to the original topic of this issue (missing unit descripitions) the `mts` branch that I'm working on in #17 has just one file for unit descriptions -- `data/unit-definitions.csv`.

If you have a suggestion for the IPMs unit and a reference to back it up with, you could offer the suggestion as a PR back into `mts`, if you're still inclined to do so. 

I'm still streamlining my tileset-updating process, but I'm in a **much** better position now to adjust descriptions and unit names than I was several months ago.

**Comment by brandon-rhodes** [comment]
Thanks for the update! It's exciting to see the flurry of PR's as you make progress on this valuable resrouce. I'll move this back nearer the top of my to-do list now that it sounds like you've found a viable way forward and will have a pipeline allowing updates to the data to flow through to the map. (I'll hopefully be in a wilderness area in Colorado next week, though, so it might be July that you see activity from me!)
